### PR TITLE
add optimizer comments

### PR DESCRIPTION
**Pull request by czpmango**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

### CODE REVIEW

File: src/graph/optimizer/rule/CollapseProjectRule.h [hunk 1]
```diff
@@ -13,16 +13,36 @@ DECLARE_bool(enable_optimizer_collapse_project_rule);
...
-/**
- * Combines two [[Project]] nodes into one
- * Required conditions:
- *  1. Match the pattern
- *  2. Expressions between nodes cannot be referenced more than once
- * Benefits:
- *  1. reduce the copy of memory between nodes
- *  2. reduces expression overhead in some cases(similar to column pruning)
- */
+/*
```
Comment (Aiee) [inline src/graph/optimizer/rule/CollapseProjectRule.h:16]: I believe we should use `//`

**Review by yixinglu — DISMISSED** [review]

File: src/graph/optimizer/rule/CollapseProjectRule.h (hunk 1 above)
Comment (czpmango) [reply src/graph/optimizer/rule/CollapseProjectRule.h:16]: Any reference?
Google style?

Comment (CPWstatic) [reply src/graph/optimizer/rule/CollapseProjectRule.h:16]: Google style doesn't declare that the comments should use `//`, but it suggests that the comments style should be unified. I think it makes sense. And the most open source projects i had ever seen use the `//`. For now i have seen much styles in our project, it's too ugly.
```
// 

/*

*/

/**
 *
 */
```

Comment (czpmango) [reply src/graph/optimizer/rule/CollapseProjectRule.h:16]: ACK.

File: src/graph/optimizer/OptContext.h [hunk 2]
```diff
@@ -48,8 +48,10 @@ class OptContext final : private boost::noncopyable, private cpp::NonMovable {
   const OptGroupNode *findOptGroupNodeByPlanNodeId(int64_t planNodeId) const;
 
  private:
+  // A global flag to record whether this iteration caused a change to the plan
   bool changed_{true};
   graph::QueryContext *qctx_{nullptr};
+  // Memo memory management in the Optimizer phase
```
Comment (jievince) [inline src/graph/optimizer/OptContext.h:54]: What is memo?

**Review by jievince — APPROVED** [review]

File: src/graph/optimizer/OptContext.h (hunk 2 above)
Comment (Aiee) [inline src/graph/optimizer/OptContext.h:54]: typo?

File: src/graph/optimizer/rule/PushLimitDownScanAppendVerticesRule.h [hunk 3]
```diff
@@ -11,6 +11,45 @@
...
+//  +--------+--------+
+//           |
+// +---------+---------+
+// |   AppendVertices  |
+// +---------+---------+
+//           |
+// +---------+---------+
+// |    ScanVertices   |
+// |     (limit=3)     |
+// +---------+---------+
```
Comment (Aiee) [inline src/graph/optimizer/rule/PushLimitDownScanAppendVerticesRule.h:51]: Will there still be a limit node after optimization?

File: src/graph/optimizer/OptContext.h (hunk 2 above)
Comment (Shylock-Hg) [inline src/graph/optimizer/OptContext.h:54]: What's this?

**Review by yixinglu — APPROVED** [review]

**Pull request merged**